functions/src/project/delete-project.ts

```diff
@@ -0,0 +1,44 @@
+// Third party modules
+import { firestore, CloudFunction, EventContext } from 'firebase-functions';
+
+// Dashboard hub firebase functions models/mappers
+import { DocumentData, DocumentSnapshot, FirebaseAdmin, WriteResult } from './../client/firebase-admin';
+
+export const onDeleteProjectRepositories: CloudFunction<DocumentSnapshot> = firestore
+  .document('projects/{projectUid}')
+  .onDelete(async (projectSnapshot: DocumentSnapshot, context: EventContext) => {
+
+    try {
+      const project: DocumentData = projectSnapshot.data();
+
+      if (Array.isArray(project.repositories) && project.repositories.length > 0) {
+
+        const promiseList: Promise<WriteResult>[] = [];
+
+        for (const repositoryUid of project.repositories) {
```

Comment:
I prefer a `forEach` if possible?

Reply:
I use `async/await`, it doesn’t work with `forEach`

Reply:
ok